Common/src/geometry_structure.cpp

```diff
@@ -3337,11 +3336,11 @@ void CGeometry::FilterValuesAtElementCG(const vector<su2double> filter_radius,
         /*--- morphology kernels (image processing) ---*/
         case DILATE_MORPH_FILTER: case ERODE_MORPH_FILTER:
             
-          for (vector<long>::iterator it=neighbours.begin(); it!=neighbours.end(); ++it)
+          for (auto idx : neighbours)
```

Comment:
We have to be careful not to overuse the `auto` keyword. It does not always help with readability.